Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingAnalyticsFragmentTest.kt

```diff
@@ -0,0 +1,103 @@
+import dagger.android.ContributesAndroidInjector
+import de.rki.coronawarnapp.datadonation.analytics.AnalyticsSettings
+import de.rki.coronawarnapp.datadonation.analytics.common.Districts
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.spyk
+import io.mockk.unmockkAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseUITest
+import testhelpers.SCREENSHOT_DELAY_TIME
+import testhelpers.Screenshot
+import testhelpers.SystemUIDemoModeRule
+import testhelpers.TestDispatcherProvider
+import testhelpers.launchFragment2
+import testhelpers.launchFragmentInContainer2
+import tools.fastlane.screengrab.Screengrab
+import tools.fastlane.screengrab.locale.LocaleTestRule
+
+@RunWith(AndroidJUnit4::class)
+class OnboardingAnalyticsFragmentTest : BaseUITest() {
+
+    @MockK lateinit var settings: AnalyticsSettings
+    @MockK lateinit var districts: Districts
+
+    private lateinit var viewModel: OnboardingAnalyticsViewModel
+
+    @Rule
+    @JvmField
+    val localeTestRule = LocaleTestRule()
+
+    @get:Rule
+    val systemUIDemoModeRule = SystemUIDemoModeRule()
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxed = true)
+
+        coEvery { districts.loadDistricts() } returns listOf(Districts.District(
+            districtId = 11011004,
+            districtName = "SK Berlin Charlottenburg-Wilmersdorf"
+        ))
+
+        viewModel = onboardingAnalyticsViewModelSpy()
+        with(viewModel) {
+            every { ageGroup } returns flowOf(PpaData.PPAAgeGroup.AGE_GROUP_0_TO_29).asLiveData()
+            every { federalState } returns flowOf(PpaData.PPAFederalState.FEDERAL_STATE_BE).asLiveData()
+            every { district } returns flow { emit(districts.loadDistricts().first()) }.asLiveData()
+        }
+
+        setupMockViewModel(
+            object : OnboardingAnalyticsViewModel.Factory {
+                override fun create(): OnboardingAnalyticsViewModel = viewModel
+            }
+        )
+    }
+
+    private fun onboardingAnalyticsViewModelSpy() = spyk(
+        OnboardingAnalyticsViewModel(
+            settings = settings,
+            districts = districts,
+            dispatcherProvider = TestDispatcherProvider()
+        )
+    )
+
+    @After
+    fun teardown() {
+        clearAllViewModels()
+        unmockkAll()
+    }
+
+    @Test
+    fun launch_fragment() {
+        launchFragment2<OnboardingAnalyticsFragment>()
+    }
+
+    @Screenshot
+    @Test
+    fun capture_screenshot() {
+        launchFragmentInContainer2<OnboardingAnalyticsFragment>()
+        Thread.sleep(SCREENSHOT_DELAY_TIME)
+        Screengrab.screenshot(OnboardingAnalyticsFragment::class.simpleName)
```

Comment:
Please use `takeScreenshot< OnboardingAnalyticsFragmen >()` or `captureScreenshot< OnboardingAnalyticsFragment >` - which will also launch the fragment and hence removes `launchFragmentInContainer2<OnboardingAnalyticsFragment>()`